File: unique.R

ptions(scipen = 999)

setwd("~/development/kaggle-talking-data/")

app.labels = read.csv("data/app_labels.csv")
nrow(app.labels)
app.labels = unique(app.labels)
nrow(app.labels)

t = table(paste(app.labels$app_id, app.labels$label_id))
t = data.frame(sort(t, decreasing = T))



app.labels.drop.dups = read.csv("data/app_labels_drop_dups.csv", sep = ";")
nrow(app.labels.drop.dups)

t.dd = table(paste(app.labels.drop.dups$app_id, app.labels.drop.dups$label_id))
t.dd = data.frame(sort(t.dd, decreasing = T))

#al = read.csv("data/app_labels.csv")
#al[al$app_id == -5210291384343299072 & al$label_id == 405,]
#app.labels.drop.dups[app.labels.drop.dups$app_id == -5210291384343299072 & app.labels.drop.dups$label_id == 405,]

File: unique.py
import pandas as pd
from os import chdir
import gc

working_dir = "/home/tales/development/kaggle-talking-data/"
chdir(working_dir)

dtypes = {'label_id': str, 'category': str, "app_id": str, "event_id": str, "app_id": str, "is_installed": str,  "is_active": str, "gender": str ,"age": int, "group": str}

app_labels = pd.read_csv("data/app_labels.csv", dtype=dtypes)
len(app_labels)
app_labels = app_labels.drop_duplicates()
len(app_labels)

app_labels.to_csv("data/app_labels_drop_dups.csv", sep = ";")
